modules/lang-painless/src/test/java/org/elasticsearch/painless/ElvisTests.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.painless;
+
+import static java.util.Collections.singletonMap;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.lessThan;
+
+/**
+ * Tests for the Elvis operator ({@code ?:}).
+ */
+public class ElvisTests extends ScriptTestCase {
+    public void testBasics() {
```

Comment:
@nik9000 Robert and I had a long conversation about boxing early during development.  We decided to eliminate it as much as possible because of serious complications involving promotion and casting (which as you know is already very complicated).  There's only a couple of places auto-boxing happens -- arguments to methods because it would be hard to force a user to cast something to an object to add to a list and with anything related to def type because .  Otherwise, there is no auto-boxing in painless.  Perhaps, this should be the same for consistency?  Sorry, I sort of missed this yesterday thinking about the cases, but def should work anyway already, otherwise primitives don't make sense here since we don't allow Integer to become an int anywhere else.